tests/basic_correctness/test_chunked_prefill.py

```diff
@@ -62,3 +77,75 @@ def test_models(
         name_0="hf",
         name_1="vllm",
     )
+
+
+@pytest.mark.parametrize("model", LOG_PROBS_MODELS)
+@pytest.mark.parametrize("kv_cache_dtype", ["fp8", "fp8_e5m2"])
+@pytest.mark.parametrize("max_tokens", [32])
+@pytest.mark.parametrize("chunked_prefill_token_size", [1, 4, 16])
+@pytest.mark.parametrize("enforce_eager", [False, True])
+# NOTE: Increasing this in this suite will fail CI because we currently cannot
+# reset distributed env properly. Use a value > 1 just when you test.
+@pytest.mark.parametrize("tensor_parallel_size", [1])
+def test_models_log_probs(
+    vllm_runner,
+    example_prompts,
+    model: str,
+    kv_cache_dtype: str,
+    max_tokens: int,
+    chunked_prefill_token_size: int,
+    enforce_eager: bool,
+    tensor_parallel_size: int,
+) -> None:
+    """
+    Only checks log probs match between chunked-prefill and
+    non-chunked-prefill version of vLLM model runner.
+    
+    This test is used when there is discrepancy in kernels
+    / numerics (e.g. when using lower-precision types like FP8).
+    """
+    if kv_cache_dtype == "fp8_e5m2" and model not in E5M2_KV_MODELS:
+        pytest.skip(
+            f"{model} not in {E5M2_KV_MODELS} requiring {kv_cache_dtype}")
+    elif "fp8" in kv_cache_dtype and model not in E4M3_KV_MODELS:
+        pytest.skip(
+            f"{model} not in {E4M3_KV_MODELS} requiring {kv_cache_dtype}")
```

Comment:
I would suggest the following organization:
1. The model list includes `MODELS` and `FP8_KV_MODELS`.
2. For `fp8_e5m2` kv-cache, simply test with `MODELS`.
3. For `fp8_e4m3` kv-cache, test with `FP8_KV_MODELS`.

So that you could have

```python
@pytest.mark.parametrize("model_n_kv_dtype", [("fp8_e5m2", MODELS), ("fp8_e4m3", FP8_KV_MODELS)])
...
def test_model_with_fp8_kv_cache(model_n_kv_dtype, ...):
    kv_cache_dtype, model = model_n_kv_dtype
    ....
```

Reply:
I think models for which the KV has not been explicitly quantized there can be significant numerical error, even though the outputs are sensible. Hence I think it is better to only test models that have been quantized

If you want I can include it to exercise the code path but it is not possible to check correctness IMO.

Reply:
I will also add quantized llama-2 KV fp8 to testing

Reply:
The only reason I proposed this is because you added opt-125m which actually tests E5M2 without scaling factors.

Reply:
See above; now it is tested with llama-2-7b e5m2